terraform-tests/mysql_test.go

```diff
@@ -49,203 +53,200 @@ var _ = Describe("mysql", Label("mysql-terraform"), func() {
+					"storage_autoscale":          true,
+					"storage_autoscale_limit_gb": 200,
+				}))
+			})
+
+			It("autoscaling should be enabled", func() {
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"max_allocated_storage": Equal(float64(200)),
+					}))
+			})
+		})
 
+		When("storage_autoscale is true and limit <= storage_gb", func() {
+			BeforeAll(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
+					"storage_autoscale":          true,
+					"storage_autoscale_limit_gb": 5,
+				}))
 			})
 
+			It("autoscaling should be disabled", func() {
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"max_allocated_storage": BeNil(),
+					}))
+			})
 		})
+
 	})
+
+	Context("security groups", func() {
+		When("no security group ids passed", func() {
+			BeforeAll(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{}))
+			})
+
+			It("should create a new one", func() {
+				Expect(UnknownValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"vpc_security_group_ids": BeTrue(),
+					}))
+				Expect(ResourceCreationForType(plan, "aws_security_group")).To(HaveLen(1))
+				Expect(AfterValuesForType(plan, "aws_security_group")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"name": Equal("csb-mysql-test-sg"),
+					}))
+			})
+		})
+
+		When("security group ids passed", func() {
+			BeforeAll(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
+					"rds_vpc_security_group_ids": "group1,group2,group3",
+				}))
+			})
+
+			It("should use the ids passed and no create new security groups", func() {
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"vpc_security_group_ids": ConsistOf("group1", "group2", "group3"),
+					}))
+				Expect(ResourceCreationForType(plan, "aws_security_group")).To(BeEmpty())
+			})
+		})
+	})
+
+	Context("maintenance_window", func() {
+		When("no window is set", func() {
+			BeforeAll(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{}))
+			})
+			It("should not be passed", func() {
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(Not(HaveKey("maintenance_window")))
+			})
+		})
+
+		When("maintainance window specified with all values", func() {
+			BeforeAll(func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
+					"maintenance_day":        "Mon",
+					"maintenance_start_hour": "01",
+					"maintenance_end_hour":   "02",
+					"maintenance_start_min":  "03",
+					"maintenance_end_min":    "04",
+				}))
+			})
+
+			It("should pass the correct window", func() {
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"maintenance_window": Equal("mon:01:03-mon:02:04"),
```

Comment:
thoughts: I wonder how `maintenance_day` has been converted to lowercase letters